test/unit/decorators/grpc.decorator.unit.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback4-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {expect} from '@loopback/testlab';
+import {grpc} from '../../..';
+import {Application} from '@loopback/core';
+import {GrpcBindings} from '../../../src/keys';
+import {
+  GreeterInterface,
+  HelloRequest,
+  HelloReply,
+} from '../../protos/greeter.proto';
+import {Reflector} from '@loopback/context';
+
+describe('@rpc decorator', () => {
+  it('defines reflection metadata for rpc method', () => {
+    class Greeter implements GreeterInterface {
+      @grpc()
+      SayHello(request: HelloRequest): HelloReply {
+        return {message: `hello ${request.name}`};
+      }
+      Helper(): boolean {
+        return true;
+      }
+    }
+    let controllerMethods: string[] = [];
+    const flags: {[key: string]: boolean} = {};
+    const app: Application = givenApplication();
+    app.controller(Greeter);
+    const controller = app.find('controllers.Greeter').pop();
+    if (controller) {
+      const proto = controller.valueConstructor.prototype;
+      controllerMethods = Object.getOwnPropertyNames(proto).filter(
+        key => key !== 'constructor' && typeof proto[key] === 'function',
+      );
+      for (const methodName of controllerMethods) {
+        flags[methodName] = Reflector.getMetadata(
+          GrpcBindings.LB_GRPC_HANDLER,
+          proto,
+          methodName,
+        );
+      }
+    }
+    expect(controllerMethods).to.eql(['SayHello', 'Helper']);
+    expect(flags.SayHello).to.be.true();
+    expect(flags.Helper).to.be.undefined();
```

Comment:
When something goes wrong here, e.g. there is a typo in `SayHello`, then these two checks won't provide enough information to troubleshoot the problem.

A better approach is to use [`deepEqual`](https://shouldjs.github.io/#should-deepequal) and test the whole object.

```ts
expect(flags).to.deepEqual({
  SayHello: true,
  // Helper: ignored
});